src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs

```diff
@@ -69,9 +70,24 @@ public void Dispose()
         protected virtual void Dispose(bool disposing)
         {
             // No managed resources to clean up, so disposing is ignored.
+            try
+            {
+                try
+                {
+                    Directory.Delete(TestDirectory, recursive: true);
+                }
+                catch (UnauthorizedAccessException)
+                {
+                    DirectoryInfo di = new DirectoryInfo(TestDirectory);
+                    foreach (FileInfo file in di.GetFiles("*", SearchOption.AllDirectories))
+                    {
+                        file.IsReadOnly = false;
```

Comment:
This try catch here can be improved even more if you make it look like the one we have in the Dispose method of the System.IO.FileSystem.AccessControl test class. In that method, I made sure all files and folders have the necessary ACL permissions for the current user to delete.

Up to you.